### PR TITLE
JobManager test suite

### PR DESCRIPTION
**Pull request by jsantos17**
[ch8169]

### CODE REVIEW

File: core/src/main/scala/fs2/job/JobManager.scala [hunk 1]
```diff
@@ -165,7 +166,8 @@ final class JobManager[F[_]: Concurrent: Timer, I, N] private (
               if (ignoreAbsence) {
                 val casF = Concurrent[F] delay {
                   val attempt = Context(Status.Running, Some(s.set(true)))
-                  attempt eq meta.putIfAbsent(id, attempt)
+
+                  Option(meta.putIfAbsent(id, attempt)).isEmpty
```
Comment (jsantos17) [inline core/src/main/scala/fs2/job/JobManager.scala:170]: `putIfAbsent` returns null and mutates `meta` when `id` is absent. The previous implementation always yielded false.

File: core/src/test/scala/fs2/job/JobManagerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+      refAfterStart mustEqual "Started"
+      refAfterRun mustEqual "Finished"
+    }
+
+    "cancel a job" in {
+      val JobId = 42
+
+      def jobStream(ref: SignallingRef[IO, String]): Stream[IO, Either[String, Int]] =
+        Stream.eval(ref.set("Started")).as(Right(1)) ++ await.as(Right(2)) ++ Stream.eval(ref.set("Working")).as(Right(3)) ++ await.as(Right(4)) ++ Stream.eval(ref.set("Finished")).as(Right(5))
```
Comment (jsantos17) [inline core/src/test/scala/fs2/job/JobManagerSpec.scala:92]: These `await`s aren't necessary for synchronization. They're meant to simulate work and give `fs2` a chance to interrupt/cancel the stream.

File: build.sbt [hunk 3]
```diff
@@ -20,8 +20,11 @@ lazy val core = project
   .in(file("core"))
   .settings(name := "fs2-job")
   .settings(
-    libraryDependencies += "co.fs2" %% "fs2-core" % "1.0.4",
+    libraryDependencies ++= Seq(
+        "co.fs2" %% "fs2-core" % "1.0.4",
+        "org.specs2" %% "specs2-core" % "4.3.4" % "test"),
```
Comment (djspiewak) [inline build.sbt:25]: Should probably update both of these while we're at it.

File: build.sbt [hunk 4]
```diff
@@ -20,8 +20,11 @@ lazy val core = project
...
 
+    addCompilerPlugin("com.olegpy" %% "better-monadic-for" % "0.2.4"),
```
Comment (djspiewak) [inline build.sbt:27]: And this

File: core/src/main/scala/fs2/job/JobManager.scala (hunk 1 above)
Comment (djspiewak) [reply core/src/main/scala/fs2/job/JobManager.scala:170]: Derp.

File: core/src/test/scala/fs2/job/JobManagerSpec.scala [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+      val job = Job[IO, Int, String, Int](JobId, stream ++ notification)
+
+      val (submitResult, ids, status) = (for {
+        mgr <- mkJobManager
+        submitResult <- Stream.eval(mgr.submit(job))
+        _ <- await
+        ids <- Stream.eval(mgr.jobIds)
+        status <- Stream.eval(mgr.status(JobId))
+      } yield (submitResult, ids, status)).compile.lastOrError.timeout(1.second).unsafeRunSync
```
Comment (djspiewak) [inline core/src/test/scala/fs2/job/JobManagerSpec.scala:58]: ⛳️ 

`unsafeRunTimed(1.second)`

File: core/src/test/scala/fs2/job/JobManagerSpec.scala (hunk 2 above)
Comment (djspiewak) [reply core/src/test/scala/fs2/job/JobManagerSpec.scala:92]: Thanks for this clarification!

**Review by djspiewak — APPROVED** [review]
Should upgrade the versions in a subsequent PR.

**Pull request merged**